controllers/controller/workspacerouting/solvers/common.go

```diff
@@ -68,50 +75,69 @@ func getDiscoverableServicesForEndpoints(endpoints map[string]controllerv1alpha1
 	return services
 }
 
-func getServicesForEndpoints(endpoints map[string]controllerv1alpha1.EndpointList, meta WorkspaceMetadata) []corev1.Service {
-	if len(endpoints) == 0 {
-		return nil
+// GetServiceForEndpoints returns a single service that exposes all endpoints of given exposure types, possibly also including the discoverable types.
+func GetServiceForEndpoints(endpoints map[string]controllerv1alpha1.EndpointList, meta WorkspaceMetadata, includeDiscoverable bool, exposureType ...devworkspace.EndpointExposure) corev1.Service {
```

Comment:
Should it return (no-op/redundant) service if none of endpoints satisfies the specified exposureType?

Reply:
Good point!